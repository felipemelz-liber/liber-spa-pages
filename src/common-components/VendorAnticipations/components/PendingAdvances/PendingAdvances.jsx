import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { FlatButton, FlatLink } from 'liber-components';
import {
  SubTitle,
  AdvancesContainer,
  Item,
  Card,
  Section,
  Row,
  FaceLabel,
  NetValue,
  ValueContainer,
  Footer,
  EyeIcon,
} from './PendingAdvances.styles';
import StatusText from '../../../StatusText/StatusText';
import { formatMoney } from '../../../../views/withdraw/moneyHandler';
import CancellationDialog from '../../../CancellationDialog/CancellationDialog';
import { CANCEL_PAGE_URL } from '../../urls';
import HideFromAdmin from '../HideFromAdmin/HideFromAdmin';
import { dispatchDataLayerEvent } from '../../../../vendor/Utils';

const statusReference = {
  signed: 'Realizando Pagamento',
  started: 'Aguardando Confirmação',
  created: 'Aguardando Confirmação',
};

const renderAdvanceCard = (
  advance,
  setDialogOpen,
  setSelectedAdvanceId,
  onDetailsClick,
  isAdmin,
) => {
  const { id, status, netValue, userHasSigned } = advance;

  const isDocumentGenerated = status !== 'created';
  const userHasToSign = ['created', 'started'].includes(status) && !userHasSigned;

  const handleConfirmClick = () => {
    dispatchDataLayerEvent('antecipacoesConfirmarAntecipacao', isAdmin);
  };

  const handleAdvanceShowClick = () => {
    dispatchDataLayerEvent('antecipacoesSolicitacoesEmAndamentoShowDoBordero', isAdmin);
  };

  return (
    <Item key={id}>
      <Card>
        <StatusText type="neutral">{statusReference[status]}</StatusText>
        <FaceLabel>Valor de Antecipação</FaceLabel>
        <Row>
          <ValueContainer>
            <NetValue>{`R$ ${formatMoney(netValue)}`}</NetValue>
          </ValueContainer>
          {!isAdmin ? (
            <a href={`/fornecedor/antecipacoes/${id}`} onClick={handleAdvanceShowClick}>
              <EyeIcon />
            </a>
          ) : (
            <EyeIcon onClick={() => onDetailsClick(id)} />
          )}
        </Row>
        {userHasToSign ? (
          <HideFromAdmin>
            <Footer>
              <FlatButton
                onClick={() => {
                  setDialogOpen(true);
                  setSelectedAdvanceId(advance.id);
                }}
              >
                CANCELAR
              </FlatButton>
              <FlatLink
                href={`/fornecedor/antecipacoes/sign?ids[]=${id}&generated=${isDocumentGenerated}`}
                onClick={handleConfirmClick}
              >
                CONFIRMAR
              </FlatLink>
            </Footer>
          </HideFromAdmin>
        ) : null}
      </Card>
    </Item>
  );
};

export const PendingAdvances = ({ pendingAdvances, onDetailsClick, isAdmin }) => {
  const [dialogOpen, setDialogOpen] = React.useState(false);
  const [selectedAdvanceId, setSelectedAdvanceId] = React.useState(null);

  const onDialogClose = () => {
    setSelectedAdvanceId(null);
    setDialogOpen(false);
  };

  const redirectToCancel = () => {
    dispatchDataLayerEvent('antecipacoesCancelarAntecipacao', isAdmin);
    window.location.href = `${CANCEL_PAGE_URL}?id=${selectedAdvanceId}`;
  };

  return pendingAdvances.length > 0 ? (
    <>
      <Section>
        <SubTitle>Solicitações em Andamento</SubTitle>
        <AdvancesContainer>
          {pendingAdvances.map(advance =>
            renderAdvanceCard(
              advance,
              setDialogOpen,
              setSelectedAdvanceId,
              onDetailsClick,
              isAdmin,
            ),
          )}
        </AdvancesContainer>
      </Section>
      <CancellationDialog open={dialogOpen} onClose={onDialogClose} onConfirm={redirectToCancel} />
    </>
  ) : null;
};

PendingAdvances.propTypes = {
  pendingAdvances: PropTypes.arrayOf(PropTypes.object),
  onDetailsClick: PropTypes.func,
  isAdmin: PropTypes.bool,
};

PendingAdvances.defaultProps = {
  pendingAdvances: [],
  onDetailsClick: () => null,
  isAdmin: false,
};

const mapStateToProps = ({ vendorAdvances: { pendingAdvances, isAdmin } }) => ({
  pendingAdvances,
  isAdmin,
});

export default connect(mapStateToProps)(PendingAdvances);
