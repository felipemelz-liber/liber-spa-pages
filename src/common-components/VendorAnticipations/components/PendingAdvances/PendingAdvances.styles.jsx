import styled from 'styled-components';

import { Card as RawCard, media } from 'liber-components';
import Icon from '@mdi/react';
import { mdiEye } from '@mdi/js';

export const EyeIcon = styled(Icon).attrs(() => ({
  size: '20px',
  color: '#4b6f85',
  path: mdiEye,
}))`
  cursor: pointer;
`;

export const SubTitle = styled.span`
  font-family: Roboto;
  font-size: 21px;
  font-weight: 500;
  font-stretch: normal;
  font-style: normal;
  line-height: 1.4;
  letter-spacing: 0.08px;
  text-align: left;
  color: #4b6f85;
  margin-bottom: 24px;
`;

export const Footer = styled.div`
  border-top: solid 1px #cdd7df;
  display: flex;
  justify-content: flex-end;
  padding-top: 8px;
  width: 100%;
`;

export const Section = styled.div`
  display: flex;
  flex-direction: column;
  margin-bottom: 56px;
`;

export const Item = styled.div`
  display: flex;
  flex-basis: 40%;
  ${media.large`
    flex-basis: 33%;
  `}
  box-sizing: border-box;
`;

export const AdvancesContainer = styled.div`
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  margin: -8px;
  box-sizing: content-box;
  width: calc(100% - 16px);
  & > ${Item} {
    padding: 8px;
  }
`;

export const Card = styled(RawCard).attrs(() => ({ version: 2 }))`
  padding: 20px 16px;
  padding-bottom: 8px;
  width: 100%;
  box-shadow: 0 3px 6px 0 rgba(0, 0, 0, 0.16);
  border: solid 1px #cdd7df;
  align-items: flex-start;
`;

export const Row = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  width: 100%;
  margin-bottom: 24px;
`;

export const ValueContainer = styled.div`
  display: flex;
  flex-direction: column;
`;

export const FaceLabel = styled.span`
  font-family: Roboto;
  font-size: 12px;
  font-weight: 500;
  font-stretch: normal;
  font-style: normal;
  line-height: 2.45;
  letter-spacing: 0.05px;
  text-align: left;
  color: #405f71;
  margin-top: 16px;
`;

export const NetValue = styled.span`
  font-family: Roboto;
  font-size: 24px;
  font-weight: normal;
  font-stretch: normal;
  font-style: normal;
  line-height: 1.22;
  letter-spacing: 0.1px;
  text-align: left;
  color: #6287a7;
`;
