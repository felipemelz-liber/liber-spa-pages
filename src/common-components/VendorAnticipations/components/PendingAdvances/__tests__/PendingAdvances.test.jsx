import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import { PendingAdvances } from '../PendingAdvances';

describe('PendingAdvances tests', () => {
  configure({ adapter: new Adapter() });

  const props = {
    pendingAdvances: [
      { id: 1, status: 'started', faceValue: 12000.32 },
      { id: 2, status: 'signed', faceValue: 12000.32 },
    ],
  };

  it('should render correctly without props', () => {
    const wrapper = shallow(<PendingAdvances />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render correctly with props', () => {
    const wrapper = shallow(<PendingAdvances {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
