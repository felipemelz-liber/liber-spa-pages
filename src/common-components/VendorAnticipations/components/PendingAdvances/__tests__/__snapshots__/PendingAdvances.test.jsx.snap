// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PendingAdvances tests should render correctly with props 1`] = `
<Fragment>
  <styled.div>
    <styled.span>
      Solicitações em Andamento
    </styled.span>
    <styled.div>
      <styled.div
        key="1"
      >
        <Styled(Cards__Card)>
          <StatusText
            type="neutral"
          >
            Aguardando Confirmação
          </StatusText>
          <styled.span>
            Valor de Antecipação
          </styled.span>
          <styled.div>
            <styled.div>
              <styled.span>
                R$ 0,00
              </styled.span>
            </styled.div>
            <a
              href="/fornecedor/antecipacoes/1"
              onClick={[Function]}
            >
              <Styled(Icon) />
            </a>
          </styled.div>
          <Connect(HideFromAdmin)>
            <styled.div>
              <StyledButtons__FlatButton
                color="primary"
                onClick={[Function]}
              >
                CANCELAR
              </StyledButtons__FlatButton>
              <StyledButtons__FlatLink
                color="primary"
                href="/fornecedor/antecipacoes/sign?ids[]=1&generated=true"
                onClick={[Function]}
              >
                CONFIRMAR
              </StyledButtons__FlatLink>
            </styled.div>
          </Connect(HideFromAdmin)>
        </Styled(Cards__Card)>
      </styled.div>
      <styled.div
        key="2"
      >
        <Styled(Cards__Card)>
          <StatusText
            type="neutral"
          >
            Realizando Pagamento
          </StatusText>
          <styled.span>
            Valor de Antecipação
          </styled.span>
          <styled.div>
            <styled.div>
              <styled.span>
                R$ 0,00
              </styled.span>
            </styled.div>
            <a
              href="/fornecedor/antecipacoes/2"
              onClick={[Function]}
            >
              <Styled(Icon) />
            </a>
          </styled.div>
        </Styled(Cards__Card)>
      </styled.div>
    </styled.div>
  </styled.div>
  <CancellationDialog
    onClose={[Function]}
    onConfirm={[Function]}
    open={false}
  />
</Fragment>
`;

exports[`PendingAdvances tests should render correctly without props 1`] = `""`;
